### PR TITLE
New +MENTOR permission for mentors

### PR DESCRIPTION
**Pull request by ZomgPonies**
Neca asked for a new +MENTOR flag that could ONLY have the following rights and was distinct from the +MOD flag:

-Answer Questions Ahelps
-Appear in adminwho under mentors

No msay, No ghosting, no viewing variables, no players panels/notes, etc.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I'd axe msay, personally--to prevent mentor buddying and metagaming; would likely allow Para to be all the more liberal in acceptance of mentor applications to, as the real power they have access to is virtually nil.

Personally, I'd go a step further: simplify adminhelp into two categories so it's not so confusing to people; I know Icy proposed this a while back, and I felt it was fairly legitimate; we almost never get inquiries for events and the bug reports are also very low.

**Comment by Markolie** [comment]
I'd like to keep bug reports, as I've used those reports numerous to fix bugs and we easily get 10 per round especially after patches.

**Comment by ZomgPonies** [comment]
For bug reports I would much much much rather direct people to comment on github as a matter of procedure. Anyways they can use any of the other category to report the bug reports. I don't mind having only 2 categories, as after using the old system for a while, it is true that 'Bug Reports' and 'Events' categories were severely under-utilized. I would also much rather regular admins be made aware of active bugs, which they werent with bug reports only going to people with +DEBUG.

**Comment by ZomgPonies** [comment]
https://developer.github.com/v3/issues/

If we want to get really fancy, it -could- be possible to add a bug report button ingame that automatically fills out a form and creates an issue on github.

File: code/setup.dm [hunk 1]
```diff
@@ -698,6 +698,7 @@ var/list/TAGGERLOCATIONS = list("Disposals",
 #define R_SOUNDS		2048
 #define R_SPAWN			4096
 #define R_MOD			8192
+#define R_MENTOR		16384
 
 #define R_MAXPERMISSION 8192 //This holds the maximum value for a permission. It is used in iteration, so keep it updated.
```
Comment (Markolie) [inline code/setup.dm]: Needs updating prior to merging.

**Comment by tigercat2000** [comment]
Ahem, I would like to voice concern at removing aghosting, view-varibles, and Msay. In my personal experience, mentors very heavily depend on VV and Aghosting to properly assess the context of a question, and actually determine the cause of the question. As for Msay, everything is logged as well as transmitted to higher admins, I don't think we should accept mentors that would abuse msay to "mentor buddying and metagaming".

**Comment by Svej** [comment]
I don't think aghost should be going either.

A mentor can help players by using aghost, in scenarios where he needs to see it for himself, like if a guy is bucklecuffed and he asks why he can't move.

**Comment by Markolie** [comment]
I think that goes against the point of them having as few permissions as possible, so that a much larger number of mentors can be promoted. If need be, perhaps, certain older mentors can keep their +MOD right (although that's for the headmins to decide).

**Comment by ZomgPonies** [comment]
Yeah since we have +MOD, we could have a 'Senior Mentor' role for trusted players who want the ability to better help players without the responsibilities of an actual staff position.

I'm putting a DNM tag on this PR until Neca and other headmins chip in and clearly define how they want the mentoring system to work.

**Comment by Earthdivine** [comment]
I would say Mentors should lose VV's, and to also remove the allseeing LOOC. That's not really all that needed for mentors as much as mods and admins need it, and if a person ever wanted to ask a question in LOOC, there's people around the person probably is asking already.

**Comment by Svej** [comment]
Deadsay and all-seeing LOOC is not really needed, but I think they should still have msay for the admins and mentors to coordinate with eachother.

**Comment by Fox-McCloud** [comment]
Huge :+1: 

This is long overdue; once merged in the hold on mentor apps can be lifted.

**Comment by ZomgPonies** [comment]
Alright fixed it so only admins+ can use maximum science on RnD console.
Regular admin verbs lose ability to respond to faxes and secrets panel, sorry trialmins, concentrate on rules.
Full admins and up get +EVENT with all the IC stuff in it.

**Comment by ZomgPonies** [comment]
Should be good to go now.

**Pull request merged**